grails5 upgrade

niravassar opened the pull request.

Commit:
grails5 upgrade

sdelamo commented on complete/build.gradle:
```diff
@@ -6,7 +6,7 @@ buildscript {
     dependencies {
         classpath "org.grails:grails-gradle-plugin:$grailsVersion"
```
this should be decoupled from grails version

sdelamo commented on complete/build.gradle:
```diff
@@ -19,7 +19,6 @@ apply plugin:"war"
-apply plugin: 'codenarc'
 
 repositories {
     mavenLocal()
```
get rid of mavenLocal()

sdelamo commented on complete/gradle.properties:
```diff
@@ -1,8 +1,11 @@
-gormVersion=7.0.2.RELEASE
-hibernateCoreVersion=5.4.0.Final
-gebVersion=3.2
+grailsVersion=5.0.0
```
this should be decoupled from grails version

sdelamo commented on complete/gradle.properties:
```diff
@@ -1,8 +1,11 @@
-hibernateCoreVersion=5.4.0.Final
-gebVersion=3.2
+grailsVersion=5.0.0
+gormVersion=7.1.0.M5
```
this should be GORM 7.1.0

sdelamo commented on initial/build.gradle:
```diff
@@ -6,7 +6,7 @@ buildscript {
     dependencies {
         classpath "org.grails:grails-gradle-plugin:$grailsVersion"
```
this should be decoupled from grails version

sdelamo commented on initial/build.gradle:
```diff
@@ -19,7 +19,6 @@ apply plugin:"war"
-apply plugin: 'codenarc'
 
 repositories {
     mavenLocal()
```
remove mavenLocal()

sdelamo commented on initial/gradle.properties:
```diff
@@ -1,8 +1,11 @@
-gormVersion=7.0.2.RELEASE
-hibernateCoreVersion=5.4.0.Final
-gebVersion=3.2
+grailsVersion=5.0.0
```
grails version should be 5.0.1

sdelamo commented on initial/gradle.properties:
```diff
@@ -1,8 +1,11 @@
-hibernateCoreVersion=5.4.0.Final
-gebVersion=3.2
+grailsVersion=5.0.0
+gormVersion=7.1.0.M5
```
this should be GORM 7.1.0

jamesfredley committed:
Modernize CI to Java 21 + Gradle 9 baseline (refs apache/grails-static-website#354)

jamesfredley committed:
Revert "Modernize CI to Java 21 + Gradle 9 baseline (refs apache/grails-static-website#354)"

This reverts commit b80d3133bfec300f5fa3f0e05fb9e4c9019aebed.

jamesfredley committed:
Update README to point at canonical guide on grails.apache.org (refs apache/grails-static-website#354)

jamesfredley committed:
Replace dead publisher CI with sample-app gradle check workflow

Delete: githubactions-build.sh
Add: .github/workflows/gradle.yml (auto-detects Java from Gradle wrapper version)

(refs apache/grails-static-website#354)
